docs/source/notes/autograd.rst

```diff
@@ -245,49 +265,51 @@ where
-the `JAX docs <https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Complex-numbers-and-differentiation>`_
-to get explanation for the negative signs in the formula.
+The above formula can also be verified to be correct for cross-domain functions functions.
 
-**What happens if I call backward() on a complex scalar?**
-*******************************************************************************
+The Vector-Jacobian Product for a function :math:`F: ℂ → ℝ`:
 
-The gradient for a complex function is computed assuming the input function is a holomorphic function.
-This is because for general :math:`ℂ → ℂ` functions, the Jacobian has 4 real-valued degrees of freedom
-(as in the `2x2` Jacobian matrix above), so we can’t hope to represent all of them with in a complex number.
-However, for holomorphic functions, the gradient can be fully represented with complex numbers due to the
-Cauchy-Riemann equations that ensure that `2x2` Jacobians have the special form of a scale-and-rotate
-matrix in the complex plane, i.e. the action of a single complex number under multiplication. And so, we can
-obtain that gradient using backward which is just a call to `vjp` with covector `1.0`.
+    .. code::
 
-The net effect of this assumption is that the partial derivatives of the imaginary part of the function
-(:math:`v(x, y)` above) are discarded for :func:`torch.autograd.backward` on a complex scalar
-(e.g., this is equivalent to dropping the imaginary part of the loss before performing a backwards).
+        def F(z):
+            x, y = real(z), imag(z)
+            s = u(x, y)
+            return s
 
-For any other desired behavior, you can specify the covector `grad_output` in :func:`torch.autograd.backward` call accordingly.
+at :math:`x + yj` can be simplified to:
 
-**How are the JVP and VJP defined for cross-domain functions?**
-***************************************************************
+    .. math::
+        VJP = 2 * grad\_out * \frac{\partial s}{\partial z^{*}}
+        :label: [3]
 
-Based on formulas above and the behavior we expect to see (going from :math:`ℂ → ℝ^2 → ℂ` should be an identity),
-we use the formula given below for cross-domain functions.
+since the Wirtinger and Conjugate Wirtinger derivative in this case are conjugate of each other and :math:`grad\_out` is
+real valued.
 
-The :math:`JVP` and :math:`VJP` for a :math:`f1: ℂ → ℝ^2` are defined as:
+The Vector-Jacobian Product for a function :math:`F: ℝ → ℂ`:
 
-    .. math:: JVP = J * \begin{bmatrix} c \\ d \end{bmatrix}
+    .. code::
 
-    .. math:: VJP = \begin{bmatrix} c & d \end{bmatrix} * J * \begin{bmatrix} 1 \\ -1j \end{bmatrix}
+        def F(z):
+            x, y = z, 0     # z = real(z)
+            s = u(x, y) + v(x, y) * 1j
+            return s
 
-The :math:`JVP` and :math:`VJP` for a :math:`f1: ℝ^2 → ℂ` are defined as:
+at :math:`x + yj` can be simplified to:
```

Comment:
This is always difficult to describe. But you cannot evaluate it at x + yj, because z is real valued.
Or should this operation be `C -> R -> C -> R` i.e. `G(F(real_to_complex(real(z)))))` (Not sure, if this matches your notation. Probably is G in your case the complete function)  ?